src/widgets/table.rs

```diff
@@ -602,6 +602,176 @@ mod tests {
         Table::new(vec![]).widths(&[Constraint::Percentage(110)]);
     }
 
+    // test how constraints interact with table column width allocation
+    mod table_column_widths {
+        use super::*;
+
+        fn get_test(
+            widths: &[Constraint],
+            available_width: u16,
+            selection_width: u16,
+        ) -> Vec<(u16, u16)> {
+            let table = Table::new(vec![]).widths(widths);
+
+            table.get_columns_widths(available_width, selection_width)
+        }
```

Comment:
Can you add a comment either in the function or as a docstring of what the return values are?

Reply:
Alternatively, maybe we can rename the function from `get_test` to `get_x_width`?

Reply:
sure, just to note it here, the return is the same as `get_columns_widths`, which is `x, width`

Reply:
I definitely had to scroll up a few times to work out what this represented too.